packages/adblocker-electron-preload/preload.ts

```diff
@@ -14,9 +14,7 @@ function getCosmeticsFiltersFirst(): string[] | null {
   return ipcRenderer.sendSync('get-cosmetic-filters-first', window.location.href);
 }
 function getCosmeticsFiltersUpdate(data: Omit<IBackgroundCallback, 'lifecycle'>) {
-  setImmediate(() => {
-    ipcRenderer.send('get-cosmetic-filters', window.location.href, data);
-  });
+  ipcRenderer.send('get-cosmetic-filters', window.location.href, data);
```

Comment:
Is this call blocking? I wonder if it can be an issue.

Reply:
no, this is asyncronous function just like window.postMessage so it won't block any process.
https://github.com/electron/electron/blob/main/docs/api/ipc-renderer.md#ipcrenderersendchannel-args

Reply:
It shouldn't be, you must be confused with `sendSync`:
![image](https://user-images.githubusercontent.com/1696674/229360257-332c6f6d-f0e7-418c-a85a-2a622c6f40f5.png)
![image](https://user-images.githubusercontent.com/1696674/229360325-d0f13a79-c98b-4f87-959f-0fbc0c6f6dbc.png)
See the docs: https://www.electronjs.org/docs/latest/api/ipc-renderer/

Reply:
Not saying it would wait for the response but maybe the call itself takes some time. I don’t have a good intuition about these Electron APIs so I prefer to ask :)